gma/src/include/firebase/gma/ump/types.h

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef FIREBASE_GMA_SRC_INCLUDE_FIREBASE_GMA_UMP_TYPES_H_
+#define FIREBASE_GMA_SRC_INCLUDE_FIREBASE_GMA_UMP_TYPES_H_
+
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "firebase/internal/platform.h"
+
+#if FIREBASE_PLATFORM_ANDROID
+#include <jni.h>
+#elif FIREBASE_PLATFORM_IOS || FIREBASE_PLATFORM_TVOS
+extern "C" {
+#include <objc/objc.h>
+}  // extern "C"
+#endif  // FIREBASE_PLATFORM_ANDROID, FIREBASE_PLATFORM_IOS,
+        // FIREBASE_PLATFORM_TVOS
+
+namespace firebase {
+namespace gma {
+namespace ump {
+
+/// Debug values for testing geography.
+enum ConsentDebugGeography {
+  /// Disable geography debugging.
+  kConsentDebugGeographyDisabled = 0,
+  /// Geography appears as in EEA for debug devices.
+  kConsentDebugGeographyEEA,
+  /// Geography appears as not in EEA for debug devices.
+  kConsentDebugGeographyNonEEA
+};
+
+/// Debug settings for `ConsentInfo::RequestConsentInfoUpdate()`. These let you
+/// force a speific geographic location. Be sure to include debug device IDs to
+/// enable this on hardware. Debug features are always enabled for simulators.
+struct ConsentDebugSettings {
+  /// Create a default debug setting, with debugging disabled.
+  ConsentDebugSettings() : debug_geography(kConsentDebugGeographyDisabled) {}
+
+  /// The geographical location, for debugging.
+  ConsentDebugGeography debug_geography;
+  /// A list of all device IDs that are allowed to use debug settings. You can
+  /// obtain this from the device log after running with debug settings enabled.
+  std::vector<std::string> debug_device_ids;
+};
+
+/// Parameters for the `ConsentInfo::RequestConsentInfoUpdate()` operation. You
+/// must explicitly set the age of consent tag (to true or false) or the
+/// operation will fail.
+class ConsentRequestParameters {
+ public:
+  ConsentRequestParameters()
+      : has_debug_settings_(false),
+        tag_for_under_age_of_consent_(false),
+        has_tag_for_under_age_of_consent_(false) {}
+  /// Set the age of consent tag. This indicates whether the user is tagged for
+  /// under age of consent. This is a required setting.
```

Comment:
I think that this does not need to be explicitly set, and defaults to false. However if it is set to true, the user will be considered under age of consent and no consent form will be available.

Reply:
Changed to a struct and removed this requirement.